src/test/testTrading.cpp

```diff
@@ -269,4 +272,40 @@ BOOST_FIXTURE_TEST_CASE(TradeFail, TradeFixture)
     testExpectedWares();
 }
 
+BOOST_FIXTURE_TEST_CASE(TradeMessages, TradeFixture)
+{
+    initGameRNG();
+    PostBox* postbox = world.GetPostMgr().AddPostBox(0);
+
+    this->TradeOverLand(players[0]->GetHQPos(), GD_NOTHING, JOB_WOODCUTTER, 2);
+    numHelpers -= 1;
+    numWoodcutters -= 2;
+    testAfterLeaving(2);
+
+    unsigned distance = world.CalcDistance(curWh->GetPos(), players[0]->GetHQPos()) + 2;
+    RTTR_SKIP_GFS(20 * distance);
+    // warriors are recruited
+    numHelpers -= numSwords;
+
+    const PostMsg* post = postbox->GetMsg(0);
+    BOOST_REQUIRE(post);
+    const PostMsgWithBuilding* msg = dynamic_cast<const PostMsgWithBuilding*>(post);
+    std::string text = boost::str(boost::format(_("Trade caravan with %s %s arrives from player '%s'.")) % 2 % JOB_NAMES[JOB_WOODCUTTER] % players[1]->name);
+    BOOST_REQUIRE_EQUAL(msg->GetText(), text);
```

Comment:
I would just check, if the text contains the number, the wares and the player instead of the exact text so it is not duplicated in 3 locations.

Reply:
i hope you suggested what i have done now.